server/src/test/java/org/elasticsearch/index/mapper/DocumentParserTests.java

```diff
@@ -138,6 +138,123 @@ public void testParseWithShadowedMultiField() throws Exception {
         assertNotNull(doc.rootDoc().getField("field.keyword"));
     }
 
+    public void testRuntimeFieldAndArrayChildren() throws IOException {
+        DocumentMapper mapper = createDocumentMapper(topMapping(b -> {
+            b.field("dynamic", "true");
+            b.startObject("runtime");
+            {
+                b.startObject("object").field("type", "test").endObject();
```

Comment:
maybe calling object is misleading given that it can end up being an array in the docs?